src/PSRule.Rules.Azure/rules/Azure.AppService.Rule.ps1

```diff
@@ -155,19 +155,68 @@ Rule 'Azure.AppService.WebProbePath' -Ref 'AZR-000080' -With 'Azure.AppService.I
         $Assert.In($siteConfig, 'Properties.ftpsState', @(
-            'FtpsOnly'
-            'Disabled'
-        ));
+                'FtpsOnly'
+                'Disabled'
+            ));
+    }
+}
+
+# Synopsis: Use supported Node.js versions for the runtime of applications.
+Rule 'Azure.AppService.MigrateNodeJs' -Ref 'AZR-000425' -Type 'Microsoft.Web/sites', 'Microsoft.Web/sites/config', 'Microsoft.Web/sites/slots', 'Microsoft.Web/sites/slots/config' -Tag @{ release = 'GA'; ruleSet = '2024_06'; 'Azure.WAF/pillar' = 'Reliability'; } {
+    
+    [SemVer[]]$versions = @(
+        # App Service on Linux. Works when main object equals Microsoft.Web/sites or Microsoft.Web/sites/slots
+        $TargetObject.properties.siteConfig.linuxFxVersion | Where-Object { $_ -like 'NODE|*' -and $_ -ne 'NODE|lts' }
+        # App Service on Linux. Works for when main object equals Microsoft.Web/sites/config 'web' or Microsoft.Web/sites/slots/config 'web'
+        $TargetObject.properties.linuxFxVersion | Where-Object { $_ -like 'NODE|*' -and $_ -ne 'NODE|lts' }
+        # App Service on Linux.
+        GetSubResources -ResourceType 'Microsoft.Web/sites/slots' | 
+        ForEach-Object { $_.properties.siteConfig.linuxFxVersion | Where-Object { $_ -like 'NODE|*' -and $_ -ne 'NODE|lts' } }
+        # App Service on Linux.
+        GetSubResources -ResourceType 'Microsoft.Web/sites/config', 'Microsoft.Web/sites/slots/config' |
+        Where-Object name -eq 'web' |
+        ForEach-Object { $_.properties.linuxFxVersion | Where-Object { $_ -like 'NODE|*' -and $_ -ne 'NODE|lts' } }
+        
+        # App Service on Windows. Works for when main object equals Microsoft.Web/sites or Microsoft.Web/sites/slots
+        $TargetObject.properties.siteConfig.appSettings.WEBSITE_NODE_DEFAULT_VERSION
+        # App Service on Windows. Works for when main object equals Microsoft.Web/sites/config 'appsettings' or Microsoft.Web/sites/slots/config 'appsettings'
+        $TargetObject.properties.WEBSITE_NODE_DEFAULT_VERSION
+        # App Service on Windows. Works for when main object equals Microsoft.Web/sites/config 'web' or Microsoft.Web/sites/slots/config 'web'
+        $TargetObject.properties.appSettings.WEBSITE_NODE_DEFAULT_VERSION
+        # App Service on Windows.
+        GetSubResources -ResourceType 'Microsoft.Web/sites/slots' |
+        ForEach-Object { $_.properties.siteConfig.appSettings.WEBSITE_NODE_DEFAULT_VERSION }
+        # App Service on Windows.
+        GetSubResources -ResourceType 'Microsoft.Web/sites/config', 'Microsoft.Web/sites/slots/config' |
+        Where-Object name -eq 'appsettings' |
+        ForEach-Object { $_.properties.WEBSITE_NODE_DEFAULT_VERSION }
+        # App Service on Windows.
+        GetSubResources -ResourceType 'Microsoft.Web/sites/config', 'Microsoft.Web/sites/slots/config' |
+        Where-Object name -eq 'web' |
+        ForEach-Object { $_.properties.appSettings.WEBSITE_NODE_DEFAULT_VERSION }
```

Comment:
Good job identifying all the possible cases.

This should be functionally fine, but if we can consider breaking it down similar to `GetWebSiteConfig` to reduce some of the complexity.

Reply:
Thanks @BernieWhite. Where is source code for `GetWebSiteConfig`? I have tried to search within this repository, but not able to find it.

Reply:
When you have the time @BernieWhite, perhaps you didn't see this.

Reply:
@BenjaminEngeset https://github.com/Azure/PSRule.Rules.Azure/blob/1e28104a65a3de731cc43557cc9ed17b24412c9c/src/PSRule.Rules.Azure/rules/Azure.AppService.Rule.ps1#L201-L210

Reply:
Thanks @BernieWhite.